executors/http/src/main/java/net/opentsdb/query/execution/HttpQueryV3Source.java

```diff
@@ -188,8 +189,20 @@ public void fetchNext(final Span span) {
     // TODO - we need to confirm the graph links.
     Map<String, QueryNodeConfig> pushdowns = Maps.newHashMap();
     pushdowns.put(config.getId(), source_builder.build());
-    for (final QueryNodeConfig pushdown : config.getPushDownNodes()) {
+    int index = 0;
+    for (QueryNodeConfig pushdown : config.getPushDownNodes()) {
+      if (pushdown instanceof DownsampleConfig) {
+        DownsampleConfig.Builder b = DownsampleConfig
+            .newBuilder((DownsampleConfig) pushdown)
+            .setStart(context.query().getStart())
+            .setEnd(context.query().getEnd())
+            .setId(pushdown.getId());
+
+        pushdown = b.build();
+        config.getPushDownNodes().set(index, pushdown);
```

Comment:
Can't do this as it's a Concurrent Modification exception. Just push it directly to the new list as per line 204.

Reply:
Guess it's ok since we're modifying the set position.